runtime/src/bank.rs

```diff
@@ -1292,6 +1334,41 @@ impl Bank {
         self.epoch_schedule.get_slots_in_epoch(prev_epoch) as f64 / self.slots_per_year
     }
 
+    fn reduce_overage_stakes(&self) {
+        // is this really all of parsable accounts owned by stake11111?
+        let mut count = 0;
+        self.cloned_stake_delegations()
+            .into_iter()
+            .for_each(|(stake_pubkey, _delegation)| {
+                if let Some(mut stake_account) = self.get_account(&stake_pubkey) {
+                    // metrics to record occurances??
+                    if let Ok(result) = stake_state::reduce_overage_stakes(
+                        &mut stake_account,
+                        &self.rent_collector.rent,
+                    ) {
+                        self.store_account(&stake_pubkey, &stake_account);
+                        let message =
+                            format!("reduced overage stake: {}, {:?}", stake_pubkey, result,);
+                        info!("{}", message);
+                        datapoint_info!("stake_info", ("info", message, String));
+                        count += 1;
+                    }
+                }
+            });
+
+        let msg = format!(
+            "bank (slot: {}): reduce_overage_stakes: {} accounts rewritten",
```

Comment:
I think a single datapoint that reports # delegations examined and # accounts rewritten would be useful.

Reply:
yeah, good idea. :)